openforcefield/topology/topology.py

```diff
@@ -0,0 +1,2190 @@
+
+    @property
+    def molecule(self):
+        """
+        Get the reference Molecule that this TopologyAtom belongs to.
+
+        Returns
+        -------
+        openforcefield.topology.molecule.Molecule
+        """
+        return self._topology_molecule.molecule
+
+
+    @property
+    def topology_atom_index(self):
+        """
+        Get the index of this atom in its parent Topology.
+
+        Returns
+        -------
+        int
+            The index of this atom in its parent topology.
+        """
+        return self._topology_molecule.atom_start_topology_index + self._atom.molecule_atom_index
+
+    @property
+    def topology_particle_index(self):
+        """
+        Get the index of this particle in its parent Topology.
+
+        Returns
+        -------
+        int
+            The index of this atom in its parent topology.
+        """
+        return self._topology_molecule.particle_start_topology_index + self._atom.molecule_particle_index
+
+
+    @property
+    def topology_bonds(self):
+        """
+        Get the TopologyBonds connected to this TopologyAtom.
+
+        Returns
+        -------
+        iterator of openforcefield.topology.topology.TopologyBonds
+        """
+
+        for bond in self._atom.bonds:
+            reference_mol_bond_index = bond.molecule_bond_index
+            yield self._topology_molecule.bond(reference_mol_bond_index)
+
+
+
+    def __eq__(self, other):
+        return ((self._atom == other._atom) and
+                (self._topology_molecule == other._topology_molecule))
+
+    #@property
+    #def bonds(self):
+    #    """
+    #    Get the Bonds connected to this TopologyAtom.
+    #
+    #    Returns
+    #    -------
+    #    iterator of openforcefield.topology.molecule.Bonds
+    #    """
+    #    for bond in self._atom.bonds:
+    #        yield bond
+
+
+    # TODO: Add all atom properties here? Or just make people use TopologyAtom.atom for that?
+
+
+
+#=============================================================================================
+# TopologyBond
+#=============================================================================================
+
+class TopologyBond(Serializable):
+    """
+
+    """
+    def __init__(self, bond, topology_molecule):
+        """
+
+        Parameters
+        ----------
+        bond : An openforcefield.topology.molecule.Bond
+            The reference atom
+        topology_molecule : An openforcefield.topology.topology.TopologyMolecule
+            The TopologyMolecule that this TopologyAtom belongs to
+        """
+        # TODO: Type checks
+        self._bond = bond
+        self._topology_molecule = topology_molecule
+
+    @property
+    def bond(self):
+        """
+        Get the reference Bond for this TopologyBond.
+
+        Returns
+        -------
+        an openforcefield.topology.molecule.Bond
+        """
+        return self._bond
+
+    @property
+    def topology_molecule(self):
```

Comment:
Again, I'd wonder whether we need to prepend these properties with `topology_` since it is a `TopologyMolecule`, and think we should tag experimental parts of the API as such in their docstrings.